HiP-DataStore/Core/HistoryUtil.cs

```diff
@@ -27,19 +34,54 @@ public static class HistoryUtil
         /// The event stream is assumed to be consistent. If the stream is inconsistent (e.g. has a create event
         /// immediately followed by another create event), the behavior and resulting summary is undefined.
         /// </remarks>
-        public static async Task<HistorySummary> GetSummaryAsync(IEventStream eventStream, EntityId entityId)
+        public static async Task<HistorySummary> GetSummaryAsync(IEventStream eventStream, EntityId entityId, string userStoreBaseUrl, DataStoreAuthConfig dataStoreAuthConfig, ILogger<HistoryController> logger)
         {
             var enumerator = eventStream.GetEnumerator();
             var summary = new HistorySummary();
+            AllItemsResultOfUserResult allUsers = null;
+            var userService = new UsersClient(userStoreBaseUrl);
+            try
+            {
+                //we should get machine to machine access token and assign it to userService.Authorization
+                string accessToken = await Auth.GetAccessTokenAsync(dataStoreAuthConfig.Domain, dataStoreAuthConfig.Audience, dataStoreAuthConfig.ClientId, dataStoreAuthConfig.ClientSecret);
+
+                userService.Authorization = "Bearer "+ accessToken;
+                //get the details of all the users, so we contact the UserStore once instead of contacting it everytime for every change                
+                allUsers = await userService.GetAllAsync(new UserQueryArgs());
+            }
+            catch (SwaggerException e)
+            {
+                logger.LogWarning(e,"The request, for getting an access token or the users' details, has been failed. The summary of changes will not show the users' names");
+            } 
+            catch (Exception e)
+            {
+                logger.LogWarning(e, "The request, for getting an access token or the users' details, has been failed. The summary of changes will not show the users' names");
```

Comment:
This exception handler will also handle SwaggerExceptions so the first catch block can be removed.

Reply:
I added the exception handling logic, so in case there is a problem obtaining the Auth0 access token or a problem obtaining the users' details from UserStore, then the application can continue and show the changes without the changers' names.

Reply:
I know, but what about getting the access token? is it going also to throw Swagger exception as well? and what if we changed the code in the future and stopped using Swagger!
The second exception handling is to catch other generic exceptions that may appear in the future other than swagger

Reply:
I think I may not have been clear enough. There are two different options how to deal with this. The first option is to log a more specific error message when a SwaggerException is thrown since only the request to the UserStore will throw such an exception. But if we want to leave the error message as it currently is, the catch block that catches the SwaggerException has no real purpose, since if you remove this catch block the other catch block (for the generic Exception) will get triggered and log the same error message. So the second option would be to remove the catch block that catches the SwaggerException

Reply:
If you agree with me, let's stick with option 1!